resources/assets/js/annotations/components/annotationCanvas/polygonBrushInteraction.vue

```diff
@@ -99,34 +99,32 @@ export default {
                 this.map.addInteraction(shiftClickSelectInteraction);
             }
         },
-    },
-    created() {
-        if (this.canAdd) {
-            Keyboard.on('e', this.togglePolygonBrush, 0, this.listenerSet);
-        }
-
-        if (this.canModify) {
-            Keyboard.on('r', this.togglePolygonEraser, 0, this.listenerSet);
-            Keyboard.on('t', this.togglePolygonFill, 0, this.listenerSet);
+        toggleShiftClickSelectInteraction(mode) {
+            shiftClickSelectInteraction.setActive(this.canModify && (mode === 'polygonEraser' || mode === 'polygonFill'));
         }
+    },
+    watch: {
+        interactionMode(mode) {
+            this.toggleShiftClickSelectInteraction(mode);
+            this.toggleCurrentInteraction(mode);
 
-        if (this.canAdd || this.canModify) {
-            this.$watch('interactionMode', this.toggleCurrentInteraction);
-        }
+        },
     },
     mounted() {
-        if (this.canModify) {
-            shiftClickSelectInteraction = new SelectInteraction({
-                condition(e) {
-                    return click(e) && shiftKeyOnly(e);
-                },
-                style: Styles.highlight,
-                layers: [this.annotationLayer],
-                features: this.selectInteraction.getFeatures(),
-                multi: true,
-            });
-            shiftClickSelectInteraction.on('select', this.handleFeatureSelect);
-        }
+        shiftClickSelectInteraction = new SelectInteraction({
+            condition(e) {
+                return click(e) && shiftKeyOnly(e);
+            },
+            style: Styles.highlight,
+            layers: [this.annotationLayer],
+            features: this.selectInteraction.getFeatures(),
+            multi: true,
+        });
+        shiftClickSelectInteraction.on('select', this.handleFeatureSelect);
+        shiftClickSelectInteraction.setActive(false);
+        Keyboard.on('r', this.togglePolygonEraser, 0, this.listenerSet);
+        Keyboard.on('t', this.togglePolygonFill, 0, this.listenerSet);
+        Keyboard.on('e', this.togglePolygonBrush, 0, this.listenerSet);
```

Comment:
Adding of the keyboard event listeners should be moved back to `created()` again.